includes/wikia/models/DesignSystemCommunityHeaderModel.class.php

```diff
@@ -475,4 +497,51 @@ private function formatLocalNavData( array $items, int $nestingLevel ): array {
 	private function getSpecialPageURL( $name ): string {
 		return SpecialPage::getTitleFor( $name )->getLocalURL();
 	}
+
+	private function getFandomStoreDataFromCache() {
+		global $wgMemc, $wgCityId;
+
+		$key = DesignSystemApiController::MEMC_PREFIX_FANDOM_STORE;
+		$memcKey = "intentX:$key:$wgCityId";
+		$cachedStoreData = $wgMemc->get( $memcKey );
+
+		return !empty( $cachedStoreData->results ) ? $this->formatFandomStoreData( $cachedStoreData->results ) : null;
+	}
+
+	private function formatFandomStoreData( $apiData ) {
+		// get copy of links from api data
+		$arrObj = new ArrayObject( $apiData );
+		$links = $arrObj->getArrayCopy();
+
+		// remove store item object from list
+		$firstItem = array_shift( $links );
+
+		// Only display 9 items + show more link
+		if ( count( $apiData ) > 10 ) {
```

Comment:
Shouldn't this check `> self::FANDOM_STORE_ITEM_LIMIT`

Reply:
I'm actually using that variable somewhere else for 9, but I see you're point, it can be resused.